homeassistant/components/izone/climate.py

```diff
@@ -0,0 +1,468 @@
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self._available
+
+    @property
+    def assumed_state(self) -> bool:
+        """Return True if unable to access real state of the entity."""
+        return not self._available
+
+    def set_available(self, available: bool) -> None:
+        """
+        Set availability for the controller.
+
+        Also sets zone availability as they follow the same availability.
+        """
+        if self.available == available:
+            return
+
+        if available:
+            _LOGGER.error(
+                "Reconnected controller %s ",
+                self._controller.device_uid)
+        else:
+            _LOGGER.error(
+                "Unable to contact controller %s",
+                self._controller.device_uid)
+
+        self._available = available
+
+    @property
+    def device_info(self):
+        """Return the device info for the iZone system."""
+        return self._device_info
+
+    @property
+    def unique_id(self):
+        """Return the ID of the controller device."""
+        return DOMAIN + '_' + self._controller.device_uid
+
+    @property
+    def name(self) -> str:
+        """Return the name of the entity."""
+        return "iZone Controller " + self._controller.device_uid
+
+    @property
+    def should_poll(self) -> bool:
+        """Return True if entity has to be polled for state.
+
+        False if entity pushes its state to HA.
+        """
+        return False
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return self._supported_features
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement which this thermostat uses."""
+        return TEMP_CELSIUS
+
+    @property
+    def precision(self):
+        """Return the precision of the system."""
+        return PRECISION_HALVES
+
+    @property
+    def state_attributes(self):
+        """Return the optional state attributes."""
+        data = super().state_attributes
+        data['supply_temperature'] = show_temp(
+            self.hass, self.supply_temperature, self.temperature_unit,
+            self.precision)
+        data['temp_setpoint'] = show_temp(
+            self.hass, self._controller.temp_setpoint,
+            self.temperature_unit, self.precision)
+        return data
+
+    @property
+    def current_operation(self):
+        """Return current operation ie. heat, cool, idle."""
+        if not self._controller.is_on:
+            return STATE_OFF
+        mode = self._controller.mode
+        for (key, value) in self._state_to_pizone.items():
+            if value == mode:
+                return key
+        assert False, "Should be unreachable"
+
+    @property
+    def operation_list(self):
+        """Return the list of available operation modes."""
+        return [STATE_OFF] + list(self._state_to_pizone.keys())
```

Comment:
```suggestion
        return [STATE_OFF, **self._state_to_pizone]
```

Just an FYI

Reply:
Nice! Didn't know you can do that with a map.